src/components/markdown-renderer/markdown-renderer.tsx

```diff
@@ -102,14 +110,44 @@ const forkAwesomeIconMap = Object.keys(ForkAwesomeIcons)
     return reduceObject
   }, {} as { [key: string]: string })
 
-export const MarkdownRenderer: React.FC<MarkdownRendererProps> = ({ content, onMetaDataChange, onFirstHeadingChange, onTocChange, className, wide }) => {
+export const MarkdownRenderer: React.FC<MarkdownRendererProps> = ({ content, onMetaDataChange, onFirstHeadingChange, onTocChange, className, wide, onLineMarkerPositionChanged }) => {
   const [tocAst, setTocAst] = useState<TocAst>()
-  const [lastTocAst, setLastTocAst] = useState<TocAst>()
+  const lastTocAst = useRef<TocAst>()
   const [yamlError, setYamlError] = useState(false)
   const rawMetaRef = useRef<RawYAMLMetadata>()
   const oldMetaRef = useRef<RawYAMLMetadata>()
   const firstHeadingRef = useRef<string>()
   const oldFirstHeadingRef = useRef<string>()
+  const documentElement = useRef<HTMLDivElement>(null)
+  const lastLineMarkerPositions = useRef<LineMarkerPosition[]>()
+
+  const calculateLineMarkerPositions = useCallback(() => {
+    if (documentElement.current && onLineMarkerPositionChanged) {
+      const lineMarkers: NodeListOf<HTMLDivElement> = documentElement.current.querySelectorAll('codimd-linemarker')
+      const lineMarkerPositions: LineMarkerPosition[] = Array.from(lineMarkers).map((marker) => {
+        return {
+          line: Number(marker.getAttribute('data-linenumber')),
+          position: marker.offsetTop
+        } as LineMarkerPosition
+      })
+      if (!equal(lineMarkerPositions, lastLineMarkerPositions.current)) {
+        lastLineMarkerPositions.current = lineMarkerPositions
+        onLineMarkerPositionChanged(lineMarkerPositions)
+      }
+    }
+  }, [onLineMarkerPositionChanged])
+
+  useEffect(() => {
+    calculateLineMarkerPositions()
+  }, [calculateLineMarkerPositions])
+
+  useResizeObserver({
+    ref: documentElement,
+    onResize: () => {
+      console.log('resize')
```

Comment:
This console.log should probably only be included in dev environments?

Reply:
Removed